### PR TITLE
Add alpine crossbuild image

### PR DESCRIPTION
**Pull request by Keno**
Probably broke the other crossbuild image, so I'm gonna have to try building that one again, but that takes some time.

### CODE REVIEW

File: workerbase/lib/build_crosscompiler.sh [hunk 1]
```diff
@@ -304,7 +312,7 @@ install_glibc_stage1()
 install_glibc_stage2()
 {
     cd /src/glibc-${glibc_version}_build
-    sudo -E chown buildworker:buildworker -R /src/glibc-${glibc_version}_build
+    sudo -E chown $(id -u):$(id -g) -R /src/glibc-${glibc_version}_build
```
Comment (staticfloat) [inline workerbase/lib/build_crosscompiler.sh:313]: While I like this in general, what was the problem here?

File: workerbase/lib/build_crosscompiler.sh [hunk 2]
```diff
@@ -335,11 +343,13 @@ install_libtapi()
     cd /src
     download_unpack.sh "${libtapi_url}"
 
+    cd /src/apple-libtapi-${libtapi_version}
+    patch -p1 < /downloads/patches/libtapi_llvm_dynlib.patch
```
Comment (staticfloat) [inline workerbase/lib/build_crosscompiler.sh]: Can you add a comment about what this patch is for?

File: workerbase/lib/build_crosscompiler.sh [hunk 3]
```diff
@@ -355,6 +365,9 @@ install_cctools()
     cd /src
     download_unpack.sh "${cctools_url}"
 
+    cd /src/cctools-port-${cctools_version}
+    patch -p1 < /downloads/patches/cctools_musl.patch
```
Comment (staticfloat) [inline workerbase/lib/build_crosscompiler.sh]: Can you add a comment about what this patch is for?

File: workerbase/lib/build_crosscompiler.sh [hunk 4]
```diff
@@ -384,6 +397,9 @@ install_dsymutil()
     cd /src
     download_unpack.sh "${dsymutil_url}"
 
+    cd /src/llvm-dsymutil-${dsymutil_version}
+    patch -p1 < /downloads/patches/dsymutil_llvm_dynlib.patch
```
Comment (staticfloat) [inline workerbase/lib/build_crosscompiler.sh]: Please add a comment about what this patch is for

File: workerbase/lib/build_crosscompiler.sh [hunk 5]
```diff
@@ -136,6 +143,8 @@ install_binutils()
 
     # Build binutils!
     cd /src/binutils-${binutils_version}
+    # NOTE: Fixed in binutils 2.29.1
+    patch -p0 < /downloads/patches/binutils-musl.patch
```
Comment (staticfloat) [inline workerbase/lib/build_crosscompiler.sh]: If this is fixed in 2.29.1, do we still need the patch?

**Review by staticfloat — CHANGES_REQUESTED** [review]

File: workerbase/lib/build_crosscompiler.sh (hunk 1 above)
Comment (Keno) [reply workerbase/lib/build_crosscompiler.sh:313]: The alpine image only has `root`.

File: workerbase/lib/build_crosscompiler.sh (hunk 5 above)
Comment (Keno) [reply workerbase/lib/build_crosscompiler.sh]: Still set to 2.28 above. Didn't want to deal with yet another change, but we can bump it and remove the patch.

Comment (staticfloat) [reply workerbase/lib/build_crosscompiler.sh]: What about https://github.com/staticfloat/julia-docker/pull/3/files#diff-e85a9af2d8ec9bcec813ba09d65e10f9R2

Comment (Keno) [reply workerbase/lib/build_crosscompiler.sh]: Ah that's for the overall binutils. We still build a binutils per target.

Comment (staticfloat) [reply workerbase/lib/build_crosscompiler.sh]: Gotcha.  Let's just bump up to `2.29.1` everywhere and have done with it.

Comment (Keno) [reply workerbase/lib/build_crosscompiler.sh]: Ok

**Pull request merged**